src/js/components/aboutTheData/TimeFilters.jsx

```diff
@@ -0,0 +1,253 @@
+                            classNames.push(`quarter-selected-${period.id}`);
+                        }
+                        return {
+                            ...period,
+                            type: classNames[classNames.length - 1],
+                            classNames: classNames.join(' '),
+                            isEnabled,
+                            component: <PeriodComponent
+                                isEnabled={isEnabled}
+                                classNames={i === 0 ? classNames.concat(['first']) : classNames}
+                                title={period.title} />
+                        };
+                    })
+            )
+        ), []);
+};
+
+const sortPeriods = ({ type: a }, { type: b }) => {
+    if (!a || !b) return 0;
+    if (a.includes('quarter-selected')) return -1;
+    if (b.includes('quarter-selected')) return 1;
+    return 0;
+};
+
+const TimePeriodFilters = ({
+    activeTab,
+    selectedPeriod,
+    selectedFy,
+    urlFy,
+    urlPeriod,
+    onTimeFilterSelection
+}) => {
+    const [dataAsOf, submissionPeriods] = useLatestAccountData();
+    const latestPeriod = getLatestPeriod(submissionPeriods);
+
+    const handleTimeChange = (fy, period, latestAvailable = latestPeriod) => {
+        onTimeFilterSelection(fy, getPeriodWithTitleById(period, latestAvailable));
+    };
+
+    useEffect(() => {
+        // when latest account data is ready or the url changes, set the active time periods
+        if (dataAsOf) {
+            const availablePeriodsInFy = submissionPeriods.filter(({ submission_fiscal_year: y }) => parseInt(urlFy, 10) === y);
+            if (availablePeriodsInFy.length) {
+                // fy selection is valid but what about the period? 🤔
+                const validPeriod = isPeriodVisible(availablePeriodsInFy, urlPeriod)
+                    ? urlPeriod
+                    : `${latestPeriod.period}`;
+
+                const selectablePeriod = isPeriodSelectable(availablePeriodsInFy, validPeriod)
+                    ? validPeriod
+                    : getLastPeriodWithinQuarterByPeriod(validPeriod);
+
+                handleTimeChange(urlFy, selectablePeriod);
+            }
+            else {
+                // invalid time selection, use the latest available fy/period 👌
+                handleTimeChange(`${latestPeriod.year}`, `${latestPeriod.period}`);
+            }
+        }
+    }, [dataAsOf, submissionPeriods, urlFy, urlPeriod]);
+
+    const generatePeriodDropdown = (fy, periods) => (
+        parsePeriods(fy, periods)
+            .map((p) => ({
+                ...p,
+                component: p.component,
+                value: `${p.id}`,
+                isEnabled: p.isEnabled,
+                onClick: p.isEnabled
+                    ? (period) => handleTimeChange(selectedFy, period)
+                    : () => console.log('dats gonna be a no from me dawg!!!')
```

Comment:
This log needs to be changed to a useful message.

Reply:
K